FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +23,45 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: url) { [weak self] result in
+			guard self != nil else { return }
+			switch result {
+			case .failure(_):
+				completion(.failure(RemoteFeedLoader.Error.connectivity))
+			case .success((let data, let httpResponse)):
+				if httpResponse.statusCode != Constants.successSatusCode {
+					completion(.failure(RemoteFeedLoader.Error.invalidData))
+				} else {
+					if let images = try? JSONDecoder().decode(FeedLoaderResponse.self, from: data) {
+						completion(.success(images.items.compactMap { $0.feedImage }))
```

Comment:
By not allowing `feedImage` to be `nil`, we can replace `compactMap` with `map`:
```suggestion
						completion(.success(images.items.map { $0.feedImage }))
```